packages/vue3-vite/src/util/utils.ts

```diff
@@ -41,28 +40,77 @@ export function updateDependencies(
   deps: Record<string, string>,
   devDeps: Record<string, string>
 ) {
-  // Make sure we don't have dependency duplicates
-  ensureDepsInPackageJson(host);
-  const depKeys = Object.keys(deps);
-  const devDepKeys = Object.keys(devDeps);
-  removeDependenciesFromPackageJson(host, depKeys, devDepKeys);
-  return addDependenciesToPackageJson(host, deps, devDeps);
+  const packageJson = ensureDepsInPackageJson(host);
+
+  // Find dependencies to be installed
+  const newDepKeys = diffArrays(
+    Object.keys(deps),
+    Object.keys(packageJson.dependencies)
+  );
+  const depsToBeInstalled = extractProps(deps, newDepKeys) as Record<
+    string,
+    string
+  >;
+
+  // Find devDependencies to be installed
+  const newDevDepKeys = diffArrays(
+    Object.keys(devDeps),
+    Object.keys(packageJson.devDependencies)
+  );
+  const devDepsToBeInstalled = extractProps(devDeps, newDevDepKeys) as Record<
+    string,
+    string
+  >;
+
+  // Add dependencies
+  return addDependenciesToPackageJson(
+    host,
+    depsToBeInstalled,
+    devDepsToBeInstalled
+  );
```

Comment:
Since we don't want the existing packages in `package.json` to be overwritten, this means we don't have to remove entries from `package.json` anymore.
Instead, we compare `deps` with the existing `packageJson.dependencies`, and only install packages that exist in `deps` but not in `packageJson.dependencies`.
The same logic applies to `devDependencies` as well.

Reply:
I simplified the code a little after investigating what `addDependenciesToPackageJson` actually does. It's a little confusing on not well-documented, but it seems like we can just get away with doing `{ ...deps, ...oldDeps }`, which will keep all the old dependencies and include new ones.